### PR TITLE
Implement feature to update password when receive requirePasswordChange with login API(#3bdgrp)

### PR DESCRIPTION
**Pull request by Yashi002**

### CODE REVIEW

File: src/views/update-password.vue [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+</template>
+
+<script>
+import {
+  IonButton,
+  IonContent,
+  IonLabel,
+  IonPage,
+  IonIcon,
+  IonInput,
```
Comment (bashu22tiwari) [inline src/views/update-password.vue:113]: We can arrange these alphabetically.

File: src/views/update-password.vue [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+    },
+  },
+  components: {
+    IonButton,
+    IonContent,
+    IonLabel,
+    IonPage,
+    IonIcon,
+    IonInput,
+    IonItem,
```
Comment (bashu22tiwari) [inline src/views/update-password.vue:244]: We can arrange these alphabetically.

File: src/views/update-password.vue (hunk 1 above)
Comment (Yashi002) [reply src/views/update-password.vue:111]: Done

File: src/views/update-password.vue (hunk 2 above)
Comment (Yashi002) [reply src/views/update-password.vue:242]: Done

File: changelogs/unreleased/-3bdgrp.yml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+---
+title: 'Implemented: logic to navigate user to update password page when requirePasswordChange
+  flag is true'
+ticket_id: "#3bdgrp"
+merge_request: 130
```
Comment (ymaheshwari1) [inline changelogs/unreleased/-3bdgrp.yml:5]: Update the MR number

File: src/i18n/locales/en.json [hunk 4]
```diff
@@ -23,6 +24,7 @@
     "Edit promise date": "Edit promise date",
     "Enter a product name, style name, SKU, UPCA or external ID.":"Enter a product name, style name, SKU, UPCA or external ID.",
     "Enter an order ID, product name, style name, SKU, customer name, UPCA or external ID":"Enter an order ID, product name, style name, SKU, customer name, UPCA or external ID",
+    "Field is required":"Field is required",
```
Comment (ymaheshwari1) [inline src/i18n/locales/en.json:27]:
```suggestion
    "Field is required": "Field is required",
```

File: src/i18n/locales/en.json [hunk 5]
```diff
@@ -38,10 +40,11 @@
     "Login": "Login",
     "Logout": "Logout",
     "Loyalty status": "Loyalty status",
+    "New password": "New password",
     "No results found":"No results found",
     "No time zone found": "No time zone found",
     "No warehouses found":"No warehouses found",
-    "OMS": "OMS",
```
Comment (ymaheshwari1) [inline src/i18n/locales/en.json:44]: We need to keep this, as we are using it on the login page.

File: src/i18n/locales/en.json [hunk 6]
```diff
@@ -51,6 +54,10 @@
     "orders": "orders",
     "Orders are released from oldest to newest. Use the Date filter to identify orders taken before a certain date.":"Orders are released from oldest to newest. Use the Date filter to identify orders taken before a certain date.",
     "Password": "Password",
+    "Password cannot be updated. Please try again.": "Password cannot be updated. Please try again.",
+    "Password must contain at least 8 characters including one digit and one special character":"Password must contain at least 8 characters including one digit and one special character",
+    "Password must match":"Password must match",
```
Comment (ymaheshwari1) [inline src/i18n/locales/en.json:59]:
```suggestion
    "Password must match": "Password must match",
```

File: src/i18n/locales/en.json [hunk 7]
```diff
@@ -51,6 +54,10 @@
     "orders": "orders",
     "Orders are released from oldest to newest. Use the Date filter to identify orders taken before a certain date.":"Orders are released from oldest to newest. Use the Date filter to identify orders taken before a certain date.",
     "Password": "Password",
+    "Password cannot be updated. Please try again.": "Password cannot be updated. Please try again.",
+    "Password must contain at least 8 characters including one digit and one special character":"Password must contain at least 8 characters including one digit and one special character",
```
Comment (ymaheshwari1) [inline src/i18n/locales/en.json:58]:
```suggestion
    "Password must contain at least 8 characters including one digit and one special character": "Password must contain at least 8 characters including one digit and one special character",
```

File: src/views/update-password.vue [hunk 8]
```diff
@@ -0,0 +1,290 @@
...
+    <ion-content :fullscreen="true">
+      <div class="flex">
+        <!-- TODO Handle enter key and prevent form submission -->
+        <v-form
+          :validation-schema="validationSchema"
+          :initialValues="formData"
+          class="update-password-container"
+          v-slot="{ validate, errors, resetForm, values }"
+        >
+          <img src="../assets/images/hc.png" />
```
Comment (ymaheshwari1) [inline src/views/update-password.vue:12]: Use Image component instead.

File: src/views/update-password.vue [hunk 9]
```diff
@@ -0,0 +1,290 @@
...
+              v-model="currentPassword"
+            >
+              <ion-input
+                name="currentPassword"
+                v-model="currentPassword"
+                :value="field.value"
+                v-bind="field"
+                :type="showPassword ? 'text' : 'password'"
+                required
+              ></ion-input>
```
Comment (ymaheshwari1) [inline src/views/update-password.vue:38]: Use self-closing tag.

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: changelogs/unreleased/-3bdgrp.yml (hunk 3 above)
Comment (Yashi002) [reply changelogs/unreleased/-3bdgrp.yml]: Updated

File: src/i18n/locales/en.json (hunk 4 above)
Comment (Yashi002) [reply src/i18n/locales/en.json]: Done

File: src/i18n/locales/en.json (hunk 5 above)
Comment (Yashi002) [reply src/i18n/locales/en.json:44]: Yes sir, done

File: src/i18n/locales/en.json (hunk 6 above)
Comment (Yashi002) [reply src/i18n/locales/en.json]: Done

File: src/i18n/locales/en.json (hunk 7 above)
Comment (Yashi002) [reply src/i18n/locales/en.json]: Done

File: src/views/update-password.vue (hunk 8 above)
Comment (Yashi002) [reply src/views/update-password.vue:12]: Image component isn't working here, with this image

File: src/views/update-password.vue (hunk 9 above)
Comment (Yashi002) [reply src/views/update-password.vue]: Done.
Thank you sir for reviewing

**Comment by k2maan** [comment]
Closing. Will be done from scratch later in #105 .

**Pull request closed without merging**